### PR TITLE
w 15 Happy thoughts API Nina Berggren

### PR DESCRIPTION
**Pull request by NinaBerggren**

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -2,25 +2,105 @@ import express from "express";
...
+
+// POST Endpoint, we get our data from JSON body. If input is valid, thought is saved 
+// and response includes thought object with _id.
+// POST handel with promisis
+app.post("/thoughts", (req, res) => {
+    const {message, createdAt} = req.body
+    // console.log(req.body);
+      const newThought =  new Thought({message: message, createdAt: createdAt}).save()
+        .then(newThought => {
+          res.status(201).json({succeess: true, response: newThought})
```
Comment (MariaWellander) [inline server.js:83]: Here's a typo on "succeess". Should be "success" :)

File: README.md [hunk 2]
```diff
@@ -1,13 +1,20 @@
...
-
-Start by briefly describing the assignment in a sentence or two. Keep it short and to the point.
+To Build our own API, with Express and MongoDB, which includes GET/POST and PATCH request endpoints to return data and to create/edit data.
 
+Using Happy thoughts project as a frontend to connect to bilt backend.
+In the Happy Thoughts project week 7 we built a frontend in React which uses an API to store thoughts in chatboard.
+Routes:
+/thougths - use either GET or POST to see a list of thoughts or post a new one.
+/thoughts/:thoughtId/like - increases the number of hearts/likes on a post.
+The GET request should only return the latest 20 results.
```
Comment (MariaWellander) [inline README.md:10]: Good explanation!

File: server.js [hunk 3]
```diff
@@ -2,25 +2,105 @@ import express from "express";
...
+// Start page
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+   res.send({
+    responseMassage: "Hello Everybody and welcome, This is a backend built for Frontend project Happy Thoughts from w7!",
+    guide: "To access the frontend use netlify Link: https://happy-thought-nina-berggren.netlify.app/ ",
+    Routes:[{
+        "/thoughts": "send GET request to see all thoughts, send POST request to create a new Happy Thought.",
+        "/thoughts/:id/like": "Updates and stor how many likes (hearths) a thought gets.",
+      }],
```
Comment (MariaWellander) [inline server.js:34]: Love this in the base path! Makes it easier to understand what the API is about.

File: server.js [hunk 4]
```diff
@@ -2,25 +2,105 @@ import express from "express";
...
+// Created Schema, the skelleton our structure, how the object suppose to look.
+
+const ThoughtSchema = new mongoose.Schema({
+  message: {
+    type: String,
+    required: true,
+    minlength: 5,
+    maxlength: 140,
+    // removes unnecessary whitespaces
+    trim: true
```
Comment (MariaWellander) [inline server.js:48]: Neat!

**Review by MariaWellander — COMMENTED** [review]
### Am I able to understand the code easily? Does the code have lots of duplicates? Could it be tidied up?

I'm able to understand the code easily and it's tidy - very good job! Some spelling errors though (so easy to make that). I'm happy to "korrläsa" if you want me to :) Love the comments and the clear and formal language throughout the code - professional!

### Does it work or does anything look broken?

I've closely tried it in and out and everything works - magnifique :)

### Does it follow the general and blue requirements?

Yes, it does!

### To sum it up

Super good work, Nina! I think that you should be very of yourself. Your first backend and frontend project - wow! As mentioned, I really like the effort of making it easy for someone else to understand the code. That's an appreciated "characteristics" out there in the wild. Really love the communication. You go!!
